certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/action/model/ActionRuleCertificateHasComplement.java

```diff
@@ -0,0 +1,17 @@
+package se.inera.intyg.certificateservice.domain.action.model;
+
+import java.util.Optional;
+import se.inera.intyg.certificateservice.domain.certificate.model.Certificate;
+
+public class ActionRuleCertificateHasComplement implements ActionRule {
```

Comment:
Suggest that you use message.type to decide if the certificate has a complement message or not. You can actual use an already existing method to do that: certificate.messages(MessageType.COMPLEMENT)

I think also it would be good if you checked that at least one of the complement messages isn't handled already or have an answer. If all are answered and/or handled, there is nothing to complement.